util/flocks/src/lib.rs

```diff
@@ -1,2 +1,37 @@
 pub mod frwlock;
-pub mod tfrwlock;
\ No newline at end of file
+pub mod tfrwlock;
+
+pub mod asynchronous {
+
+    use rustix::{
+        fd::AsFd,
+        fs::{flock as sync_flock, FlockOperation}
+    };
+    use thiserror::Error;
+
+    #[derive(Debug, Error)]
+    pub enum AsyncFlockError {
+        #[error("Lock is not available")]
+        JoinError(#[from] tokio::task::JoinError),
+        #[error("File error: {0}")]
+        IOError(#[from] rustix::io::Errno),
```

Comment:
```suggestion
        Io(#[from] rustix::io::Errno),
```

Pedantically, camel-case style only capitalizes the first letter in acronyms; IIRC std and tokio has similar variants.